src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs

```diff
@@ -191,6 +191,32 @@ private static GitPullRequest CreatePullRequest(string sourceBranch, string targ
                     select build).FirstOrDefault();
         }
 
+        internal static async Task<string> GetBuildDirectoryAsync(Build build, CancellationToken cancellationToken)
+        {
+            // used for local testing:
+            if (Options.BuildDropPath.EndsWith(@"Binaries\Debug", StringComparison.OrdinalIgnoreCase) ||
+                Options.BuildDropPath.EndsWith(@"Binaries\Release", StringComparison.OrdinalIgnoreCase))
+            {
+                return Options.BuildDropPath;
+            }
+
+            var buildClient = ProjectCollection.GetClient<BuildHttpClient>();
+
+            // Pull the build directory from the API
+            var artifacts = await buildClient.GetArtifactsAsync(build.Project.Id, build.Id, cancellationToken);
+            foreach (var artifact in artifacts)
+            {
+                // The drop path we're looking for is named the same as the build number
+                if (artifact.Name == build.BuildNumber)
+                {
+                    return Path.Combine(artifact.Resource.Data, build.BuildNumber);
```

Comment:
Will you add a note here that `artifact.Resource.Data` will always have a value because of line 101 above?  I can see myself changing line 101 and wondering why this part might fail.

Reply:
Absolutely